modules/command-engine/scaladsl/src/main/scala/surge/scaladsl/command/AggregateRef.scala

```diff
@@ -5,35 +5,31 @@ package surge.scaladsl.command
 import akka.actor.ActorRef
 import io.opentracing.Tracer
 import surge.internal.persistence.{ AggregateRefTrait, PersistentActor }
+import surge.scaladsl.common.{ AggregateRefBaseTrait, _ }
 
 import scala.concurrent.{ ExecutionContext, Future }
 
 trait AggregateRef[Agg, Cmd, Event] {
-  def sendCommand(command: Cmd)(implicit ec: ExecutionContext): Future[CommandResult[Agg]]
-  def getState(implicit ec: ExecutionContext): Future[Option[Agg]]
-  def applyEvent(event: Event)(implicit ec: ExecutionContext): Future[ApplyEventResult[Agg]]
+  def getState: Future[Option[Agg]]
+  def sendCommand(command: Cmd): Future[CommandResult[Agg]]
+  def applyEvent(event: Event): Future[ApplyEventResult[Agg]]
 }
 
-final class AggregateRefImpl[AggId, Agg, Cmd, Event](val aggregateId: AggId, val region: ActorRef, val tracer: Tracer)
+final class AggregateRefImpl[AggId, Agg, Cmd, Event](val aggregateId: AggId, protected val region: ActorRef, protected val tracer: Tracer)
     extends AggregateRef[Agg, Cmd, Event]
+    with AggregateRefBaseTrait[AggId, Agg, Cmd, Event]
     with AggregateRefTrait[AggId, Agg, Cmd, Event] {
 
-  def sendCommand(command: Cmd)(implicit ec: ExecutionContext): Future[CommandResult[Agg]] = {
+  private implicit val ec: ExecutionContext = ExecutionContext.global
+
+  def sendCommand(command: Cmd): Future[CommandResult[Agg]] = {
     val envelope = PersistentActor.ProcessMessage[Cmd](aggregateId.toString, command)
-    sendCommandWithRetries(envelope).map {
+    val result = sendCommandWithRetries(envelope).map {
```

Comment:
Maybe a leftover from debugging but we just return the result immediately after so it doesn't look like we need to assign to a variable